.github/workflows/acctest-terraform-lint.yml

```diff
@@ -40,6 +40,7 @@ jobs:
             | grep -v resource_aws_kinesis_stream_test.go \
             | grep -v resource_aws_kms_grant_test.go \
             | grep -v resource_aws_quicksight_user_test.go \
+            | grep -v resource_aws_route_table_test.go \
```

Comment:
Nit: Would be nice to avoid adding to these exclude lists so we can get rid of the custom scripting. If you need help adjusting the configurations so they pass, please reach out.

Reply:
Will do. I _think_ this was because of the dynamic attribute names so that the configuration looks like

```hcl
  %[1]s = %[2]q
```

I'll revisit.

Reply:
```console
$ find ./aws -type f -name '*_test.go' \
>             | sort -u \
>             | grep -v resource_aws_apigatewayv2_domain_name_test.go \
>             | grep -v resource_aws_ecs_capacity_provider_test.go \
>             | grep -v resource_aws_efs_file_system_test.go \
>             | grep -v resource_aws_kinesis_stream_test.go \
>             | grep -v resource_aws_kms_grant_test.go \
>             | grep -v resource_aws_quicksight_user_test.go \
>             | grep -v resource_aws_s3_bucket_object_test.go \
>             | grep -v resource_aws_sns_platform_application_test.go \
>             | xargs -I {} terrafmt diff --check --fmtcompat {}
./aws/resource_aws_route_table_test.go:1585
 data "aws_ec2_local_gateways" "all" {}
 
 data "aws_ec2_local_gateway" "first" {
   id = tolist(data.aws_ec2_local_gateways.all.ids)[0]
 }
 
 data "aws_ec2_local_gateway_route_tables" "all" {}
 
 data "aws_ec2_local_gateway_route_table" "first" {
   local_gateway_route_table_id = tolist(data.aws_ec2_local_gateway_route_tables.all.ids)[0]
 }
 
 resource "aws_vpc" "test" {
   cidr_block = "10.0.0.0/16"
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_subnet" "test" {
-  cidr_block        = "10.0.0.0/24"
-  vpc_id            = aws_vpc.test.id
+  cidr_block = "10.0.0.0/24"
+  vpc_id     = aws_vpc.test.id
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_ec2_local_gateway_route_table_vpc_association" "example" {
   local_gateway_route_table_id = data.aws_ec2_local_gateway_route_table.first.id
   vpc_id                       = aws_vpc.test.id
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_route_table" "test" {
   vpc_id = aws_vpc.test.id
 
   route {
     cidr_block       = %[2]q
     local_gateway_id = data.aws_ec2_local_gateway.first.id
   }
 
   tags = {
     Name = %[1]q
   }
 
   depends_on = [aws_ec2_local_gateway_route_table_vpc_association.example]
 }
./aws/resource_aws_route_table_test.go:1747
 resource "aws_vpc" "test" {
   cidr_block                       = "10.1.0.0/16"
   assign_generated_ipv6_cidr_block = true
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_subnet" "test" {
-  cidr_block        = "10.1.1.0/24"
-  vpc_id            = aws_vpc.test.id
-  ipv6_cidr_block   = cidrsubnet(aws_vpc.test.ipv6_cidr_block, 8, 1)
+  cidr_block      = "10.1.1.0/24"
+  vpc_id          = aws_vpc.test.id
+  ipv6_cidr_block = cidrsubnet(aws_vpc.test.ipv6_cidr_block, 8, 1)
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_network_interface" "test" {
   subnet_id = aws_subnet.test.id
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_route_table" "test" {
   vpc_id = aws_vpc.test.id
 
   route {
     ipv6_cidr_block      = %[2]q
     network_interface_id = aws_network_interface.test.id
   }
 
   tags = {
     Name = %[1]q
   }
 }
ERRO[2020-12-04 16:00:36] block 23 @ ./aws/resource_aws_route_table_test.go:1872 failed to process with: failed to parse hcl: ./aws/resource_aws_route_table_test.go:68,5-6: Invalid argument name; Argument names must not be quoted.
resource "aws_vpc" "test" {
  cidr_block                       = "10.1.0.0/16"
  assign_generated_ipv6_cidr_block = true

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_vpc" "target" {
  cidr_block = "10.0.0.0/16"

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_subnet" "test" {
  cidr_block        = "10.1.1.0/24"
  vpc_id            = aws_vpc.test.id
  availability_zone = data.aws_availability_zones.available.names[0]

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_egress_only_internet_gateway" "test" {
  vpc_id = aws_vpc.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_internet_gateway" "test" {
  vpc_id = aws_vpc.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_instance" "test" {
  ami           = data.aws_ami.amzn-ami-nat-instance.id
  instance_type = data.aws_ec2_instance_type_offering.available.instance_type
  subnet_id     = aws_subnet.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_vpc_peering_connection" "test" {
  vpc_id      = aws_vpc.test.id
  peer_vpc_id = aws_vpc.target.id
  auto_accept = true

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_route_table" "test" {
  vpc_id = aws_vpc.test.id

  route {
    "@@_@@ TFMT:%[2]s:TFMT @@_@@" = "@@_@@ TFMT:%[3]q:TFMT @@_@@"
    "@@_@@ TFMT:%[4]s:TFMT @@_@@" = "@@_@@ TFMT:%[5]s.id:TFMT @@_@@"
  }

  route {
    "@@_@@ TFMT:%[6]s:TFMT @@_@@" = "@@_@@ TFMT:%[7]q:TFMT @@_@@"
    "@@_@@ TFMT:%[8]s:TFMT @@_@@" = "@@_@@ TFMT:%[9]s.id:TFMT @@_@@"
  }

  route {
    "@@_@@ TFMT:%[10]s:TFMT @@_@@" = "@@_@@ TFMT:%[11]q:TFMT @@_@@"
    "@@_@@ TFMT:%[12]s:TFMT @@_@@" = "@@_@@ TFMT:%[13]s.id:TFMT @@_@@"
  }

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}
```

Reply:
The issue is with the escape syntax being used for identifiers which isn't compliant with https://www.terraform.io/docs/configuration/syntax.html#identifiers, https://github.com/hashicorp/hcl2/blob/master/hcl/hclsyntax/spec.md#identifiers, http://unicode.org/reports/tr31/.

Reply:
Now that Terraform supports [`null`](https://www.terraform.io/docs/configuration/expressions/types.html#null) I think I can refactor the configuration to avoid using a Go `fmt` verb in the identifier position...

Reply:
After refactoring the configuration to use conditionals with `null` there will still be an error:

```console
$ find ./aws -type f -name '*_test.go' \
>              | sort -u \
>              | grep -v resource_aws_apigatewayv2_domain_name_test.go \
>              | grep -v resource_aws_ecs_capacity_provider_test.go \
>              | grep -v resource_aws_efs_file_system_test.go \
>              | grep -v resource_aws_kinesis_stream_test.go \
>              | grep -v resource_aws_kms_grant_test.go \
>              | grep -v resource_aws_quicksight_user_test.go \
>              | grep -v resource_aws_s3_bucket_object_test.go \
>              | grep -v resource_aws_sns_platform_application_test.go \
>              | xargs -I {} terrafmt diff --check --fmtcompat {}
ERRO[2020-12-06 17:33:59] block 23 @ ./aws/resource_aws_route_table_test.go:1872 failed to process with: failed to parse hcl: ./aws/resource_aws_route_table_test.go:69,48-49: Unbalanced parentheses; Expected a closing parenthesis to terminate the expression., and 32 other diagnostic(s)
resource "aws_vpc" "test" {
  cidr_block                       = "10.1.0.0/16"
  assign_generated_ipv6_cidr_block = true

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_vpc" "target" {
  cidr_block = "10.0.0.0/16"

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_subnet" "test" {
  cidr_block        = "10.1.1.0/24"
  vpc_id            = aws_vpc.test.id
  availability_zone = data.aws_availability_zones.available.names[0]

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_egress_only_internet_gateway" "test" {
  vpc_id = aws_vpc.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_internet_gateway" "test" {
  vpc_id = aws_vpc.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_instance" "test" {
  ami           = data.aws_ami.amzn-ami-nat-instance.id
  instance_type = data.aws_ec2_instance_type_offering.available.instance_type
  subnet_id     = aws_subnet.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_vpc_peering_connection" "test" {
  vpc_id      = aws_vpc.test.id
  peer_vpc_id = aws_vpc.target.id
  auto_accept = true

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_route_table" "test" {
  vpc_id = aws_vpc.test.id

  route {
    # Destination.
    cidr_block      = (TFFMTKTBRACKETPERCENT[2]q == "cidr_block") ? %[3]q : null
    ipv6_cidr_block = (TFFMTKTBRACKETPERCENT[2]q == "ipv6_cidr_block") ? %[3]q : null

    # Target
    egress_only_gateway_id    = (TFFMTKTBRACKETPERCENT[4]q == "egress_only_gateway_id") ? %[5]s.id : null
    gateway_id                = (TFFMTKTBRACKETPERCENT[4]q == "gateway_id") ? %[5]s.id : null
    instance_id               = (TFFMTKTBRACKETPERCENT[4]q == "instance_id") ? %[5]s.id : null
    local_gateway_id          = (TFFMTKTBRACKETPERCENT[4]q == "local_gateway_id") ? %[5]s.id : null
    nat_gateway_id            = (TFFMTKTBRACKETPERCENT[4]q == "nat_gateway_id") ? %[5]s.id : null
    network_interface_id      = (TFFMTKTBRACKETPERCENT[4]q == "network_interface_id") ? %[5]s.id : null
    transit_gateway_id        = (TFFMTKTBRACKETPERCENT[4]q == "transit_gateway_id") ? %[5]s.id : null
    vpc_endpoint_id           = (TFFMTKTBRACKETPERCENT[4]q == "vpc_endpoint_id") ? %[5]s.id : null
    vpc_peering_connection_id = (TFFMTKTBRACKETPERCENT[4]q == "vpc_peering_connection_id") ? %[5]s.id : null
  }

  route {
    # Destination.
    cidr_block      = (TFFMTKTBRACKETPERCENT[6]q == "cidr_block") ? %[7]q : null
    ipv6_cidr_block = (TFFMTKTBRACKETPERCENT[6]q == "ipv6_cidr_block") ? %[7]q : null

    # Target
    egress_only_gateway_id    = (TFFMTKTBRACKETPERCENT[8]q == "egress_only_gateway_id") ? %[9]s.id : null
    gateway_id                = (TFFMTKTBRACKETPERCENT[8]q == "gateway_id") ? %[9]s.id : null
    instance_id               = (TFFMTKTBRACKETPERCENT[8]q == "instance_id") ? %[9]s.id : null
    local_gateway_id          = (TFFMTKTBRACKETPERCENT[8]q == "local_gateway_id") ? %[9]s.id : null
    nat_gateway_id            = (TFFMTKTBRACKETPERCENT[8]q == "nat_gateway_id") ? %[9]s.id : null
    network_interface_id      = (TFFMTKTBRACKETPERCENT[8]q == "network_interface_id") ? %[9]s.id : null
    transit_gateway_id        = (TFFMTKTBRACKETPERCENT[8]q == "transit_gateway_id") ? %[9]s.id : null
    vpc_endpoint_id           = (TFFMTKTBRACKETPERCENT[8]q == "vpc_endpoint_id") ? %[9]s.id : null
    vpc_peering_connection_id = (TFFMTKTBRACKETPERCENT[8]q == "vpc_peering_connection_id") ? %[9]s.id : null
  }

  route {
    # Destination.
    cidr_block      = (TFFMTKTBRACKETPERCENT[10]q == "cidr_block") ? %[11]q : null
    ipv6_cidr_block = (TFFMTKTBRACKETPERCENT[10]q == "ipv6_cidr_block") ? %[11]q : null

    # Target
    egress_only_gateway_id    = (TFFMTKTBRACKETPERCENT[12]q == "egress_only_gateway_id") ? %[13]s.id : null
    gateway_id                = (TFFMTKTBRACKETPERCENT[12]q == "gateway_id") ? %[13]s.id : null
    instance_id               = (TFFMTKTBRACKETPERCENT[12]q == "instance_id") ? %[13]s.id : null
    local_gateway_id          = (TFFMTKTBRACKETPERCENT[12]q == "local_gateway_id") ? %[13]s.id : null
    nat_gateway_id            = (TFFMTKTBRACKETPERCENT[12]q == "nat_gateway_id") ? %[13]s.id : null
    network_interface_id      = (TFFMTKTBRACKETPERCENT[12]q == "network_interface_id") ? %[13]s.id : null
    transit_gateway_id        = (TFFMTKTBRACKETPERCENT[12]q == "transit_gateway_id") ? %[13]s.id : null
    vpc_endpoint_id           = (TFFMTKTBRACKETPERCENT[12]q == "vpc_endpoint_id") ? %[13]s.id : null
    vpc_peering_connection_id = (TFFMTKTBRACKETPERCENT[12]q == "vpc_peering_connection_id") ? %[13]s.id : null
  }

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}
```

Reply:
Cool yeah, it also might be easier to just encode the arguments directly rather than formatting them in as well, even if it means creating multiple configurations or using variables/locals, but that is something that can be done later.

Reply:
I also updated `.github/workflows/acctest-terraform-lint.yml` in #16219. Once #14014 is merged and #16219 is rebased I'll try and modify the new test configuration(s) that require this.

Reply:
Refactored using local variable and `dynamic` block so that we no longer need the exclusion: https://github.com/hashicorp/terraform-provider-aws/pull/16735.